specs/composition/common/schemas/SearchResponse.yml

```
@@ -33,3 +34,227 @@ searchHits:
         $ref: 'Hit.yml#/hit'
   required:
     - hits
+
+resultsCompositionsResponse:
+  type: object
+  additionalProperties: true
+  properties:
+    compositions:
+      title: resultsCompositionsInfoResponse
+      type: object
+      additionalProperties:
+        title: resultsCompositionInfoResponse
+        type: object
+        properties:
+          injectedItems:
+            type: array
+            items:
+              $ref: '#/resultsInjectedItemInfoResponse'
+        required:
+          - injectedItems
+  required:
+    - compositions
+
+resultsInjectedItemInfoResponse:
+  type: object
+  additionalProperties: true
+  properties:
+    key:
+      type: string
+    appliedRules:
+      type: array
+      items:
+        title: resultsInjectedItemAppliedRulesInfoResponse
+        type: object
+        properties:
+          objectID:
+            $ref: '../../../common/parameters.yml#/objectID'
+        required:
+          - objectID
+  required:
+    - key
+
+# copy of specs/common/schemas/SearchResponse.yml/#baseSearchResponse without appliedRules
+compositionBaseSearchResponse:
+  type: object
+  additionalProperties: true
+  required:
+    - processingTimeMS
+  properties:
+    compositions:
+      $ref: '#/compositionsSearchResponse'
+    abTestID:
+      type: integer
+      description: A/B test ID. This is only included in the response for indices that are part of an A/B test.
```

```suggestion
    abTestID:
    abTestID:
      $ref: '../../../common/schemas/SearchResponse.yml#/baseSearchResponse/properties/abTestID'
```

Can you use references here please instead of copying? this way we still have to maintain a single object

---

for the whole baseSearchReponse ?
I couldn't make it work as it broke both the javascript & swift builds unfortunately

---

for this composition searchResponse yes you can use property level references and it should work the same way 🤔

---

ha, I'll try that, I wasn't aware this is possible

---

yup exactly like in my example should work